spec/features/results/response_form_tree_handling_spec.rb

```diff
@@ -94,8 +100,92 @@
     end
   end
 
+  describe "form submission" do
+    scenario "submitting response" do
+      visit new_hierarchical_response_path(params)
+
+      select2(user.name, from: "response_user_id")
+      fill_in("response_root_children_0_children_0_value", with: "1")
+      drop_in_dropzone(Rails.root.join("spec", "fixtures", "media", "audio", "powerup.mp3"))
+      expect(page).to have_content("The uploaded file was not an accepted format.")
+      drop_in_dropzone(Rails.root.join("spec", "fixtures", "media", "images", "the_swing.jpg"))
+      select("Animal")
+      select("Dog")
+      fill_in("response_root_children_3_children_0_children_0_children_0_value", with: "4561")
+      fill_in("response_root_children_3_children_0_children_1_children_0_children_0_value", with: "7891")
+      all("a.add-repeat").last.click
+      fill_in("response_root_children_3_children_1_children_0_children_0_value", with: "4562")
```

Comment:
I think it will be helpful to have helper methods for all this.

So like `fill_in_question(3,1,0,0, with: "blah")` and the method knows how to fill the various types. Then same style with `expect_value`